library/std/src/sys/theseus/os.rs

```diff
@@ -60,32 +82,51 @@ impl StdError for JoinPathsError {
 }
 
 pub fn current_exe() -> io::Result<PathBuf> {
-    unsupported()
+    let task = current_task()?;
+    let app_crate = task
+        .app_crate
+        .as_ref()
+        .ok_or_else(|| io_err("task didn't contain reference to app crate"))?;
```

Comment:
this would be a "BUG" right? should indicate that in the err msg i think.

Reply:
Not necessarily. If a library were to use `current_exe` then the task shouldn't contain a reference to the app crate.

Reply:
i'm not sure how a library would run without there being an executable that invoked it, but i suppose it doesn't matter either way.